src/server/page_server.cpp

```diff
@@ -60,55 +59,61 @@ page_server::set_active_tran_server_connection (cubcomm::channel &&chn)
 					std::bind (&page_server::receive_data_page_fetch, std::ref (*this),
 					    std::placeholders::_1));
 
+  m_ats_conn->register_request_handler (ats_to_ps_request::SEND_DATA_PAGE_FETCH,
+					std::bind (&page_server::receive_data_page_fetch, std::ref (*this),
+					    std::placeholders::_1));
+
   m_ats_conn->start_thread ();
 
   m_ats_request_queue.reset (new active_tran_server_request_queue (*m_ats_conn));
   m_ats_request_autosend.reset (new active_tran_server_request_autosend (*m_ats_request_queue));
   m_ats_request_autosend->start_thread ();
 }
 
-void
-page_server::disconnect_active_tran_server ()
+void page_server::disconnect_active_tran_server ()
 {
   m_ats_request_autosend.reset (nullptr);
   m_ats_request_queue.reset (nullptr);
   m_ats_conn.reset (nullptr);
 }
 
-bool
-page_server::is_active_tran_server_connected () const
+bool page_server::is_active_tran_server_connected () const
 {
   assert_page_server_type ();
 
   return m_ats_conn != nullptr;
 }
 
-void
-page_server::receive_log_prior_list (cubpacking::unpacker &upk)
+void page_server::receive_log_prior_list (cubpacking::unpacker &upk)
 {
   std::string message;
   upk.unpack_string (message);
   log_Gl.m_prior_recver.push_message (std::move (message));
 }
 
-void
-page_server::receive_log_page_fetch (cubpacking::unpacker &upk)
+void page_server::receive_log_page_fetch (cubpacking::unpacker &upk)
 {
-  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_LOG_PAGE))
-    {
-      LOG_PAGEID pageid;
-      std::string message;
+  LOG_PAGEID pageid;
+  std::string message;
 
-      upk.unpack_string (message);
-      std::memcpy (&pageid, message.c_str (), sizeof (pageid));
-      assert (message.size () == sizeof (pageid));
+  upk.unpack_string (message);
+  std::memcpy (&pageid, message.c_str (), sizeof (pageid));
+  assert (message.size () == sizeof (pageid));
 
+  if (prm_get_bool_value (PRM_ID_ER_LOG_READ_LOG_PAGE))
+    {
       _er_log_debug (ARG_FILE_LINE, "Received request for log from Transaction Server. Page ID: %lld \n", pageid);
     }
+
+  if (!m_log_page_fetcher)
+    {
+      m_log_page_fetcher.reset (new cublog::async_page_fetcher ());
+    }
```

Comment:
Use initialize/finalize functions to set/reset m_log_page_fetcher. Replace the initialization here with an assertion.
```suggestion
  assert (m_log_page_fetcher != nullptr);
```